monitor/main.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python
+
+from sets import Set
+
+import logging
+import os
+import time
+
+from prometheus_client import start_http_server, Gauge
+from kubernetes import client, config
+from kubernetes.client import ApiClient, Configuration
+from openshift.dynamic import DynamicClient
+
+MACHINE_STATUS = Gauge('machine_api_status',"Status of Machine CRs", labelnames=['machine_name','node_name','instance_state','instance_id'])
+
+# A list (implemented as a Set) of all active Machines
+ACTIVE_MACHINES = Set([])
+
+def get_machines(dynamic_client,namespace):
+    """Gets all of the Machine objects from the cluster from the specified namespace
+    """
+    machines = dynamic_client.resources.get(kind='Machine')
+    return machines.get(namespace=namespace).items
+
+def collect(dynamic_client, namespace):
+    """
+    Collect the current data from the AWS API
+    """
+
+    # List of volumes that we've actually had data back for the API
+    seen_machines = Set([])
+    machines = get_machines(dynamic_client, namespace)
+    for machine in machines:
+        seen_machines.add(machine['metadata']['name'])
+        ACTIVE_MACHINES.add(machine['metadata']['name'])
+
+        node_name = ""
+        instanceid = ""
+        value = 1
+        if 'nodeRef' in machine['status'].keys():
+            node_name = machine['status']['nodeRef']['name']
+        else:
+            value = 0
+
+        if 'instanceId' in machine['status']['providerStatus'].keys():
+            instanceid = machine['status']['providerStatus']['instanceId']
+        else:
+            value = 0
+        MACHINE_STATUS.labels(
+            machine_name   = machine['metadata']['name'],
+            node_name      = node_name,
+            instance_state = machine['status']['providerStatus']['instanceState'],
```

Comment:
Same as comment below for cleaning up inactive machine, `instanceState` won't be set if there is no instance.